src/coreComponents/mesh/ParticleSubRegion.hpp

```diff
@@ -0,0 +1,299 @@
+   */
+  array4d< real64 > & dNdX()
+  { return m_dNdX; }
+
+  /**
+   * @brief @return The array of shape function derivatives.
+   */
+  arrayView4d< real64 const > dNdX() const
+  { return m_dNdX; }
+
+  /**
+   * @brief @return The array of jacobian determinantes.
+   */
+  array2d< real64 > & detJ()
+  { return m_detJ; }
+
+  /**
+   * @brief @return The array of jacobian determinantes.
+   */
+  arrayView2d< real64 const > detJ() const
+  { return m_detJ; }
+
+  /**
+   * @brief @return The sorted array of local fractured elements.
+   */
+  SortedArray< localIndex > & fracturedElementsList()
+  { return m_fracturedCells; }
+
+  /**
+   * @brief @return The sorted array view of local fractured elements.
+   */
+  SortedArrayView< localIndex const > const fracturedElementsList() const
+  { return m_fracturedCells.toViewConst(); }
+
+  /**
+   * @brief @return The map to the embedded surfaces
+   */
+  EmbSurfMapType & embeddedSurfacesList() { return m_toEmbeddedSurfaces; }
+
+  /**
+   * @brief @return The map to the embedded surfaces
+   */
+  EmbSurfMapType const & embeddedSurfacesList() const { return m_toEmbeddedSurfaces; }
+
+  /**
+   * @brief Compute the center of each element in the subregion.
+   * @param[in] X an arrayView of (const) node positions
+   */
+  void calculateElementCenters( arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const & X ) const
+  {
+    arrayView2d< real64 > const & elementCenters = m_elementCenter;
+    localIndex nNodes = numNodesPerElement();
+
+    forAll< parallelHostPolicy >( size(), [=]( localIndex const k )
+    {
+      LvArray::tensorOps::copy< 3 >( elementCenters[ k ], X[ m_toNodesRelation( k, 0 ) ] );
+      for( localIndex a = 1; a < nNodes; ++a )
+      {
+        LvArray::tensorOps::add< 3 >( elementCenters[ k ], X[ m_toNodesRelation( k, a ) ] );
+      }
+
+      LvArray::tensorOps::scale< 3 >( elementCenters[ k ], 1.0 / nNodes );
+    } );
+  }
+
+  /// Map used for constitutive grouping
+  map< string, localIndex_array > m_constitutiveGrouping;
+
+  /// Array of constitutive point volume fraction
+  array3d< real64 > m_constitutivePointVolumeFraction;
+
+private:
+
+  /// The array of shape function derivaties.
+  array4d< real64 > m_dNdX;
+
+  /// The array of jacobian determinantes.
+  array2d< real64 > m_detJ;
+
+  /// Map of unmapped global indices in the element-to-node map
+  map< localIndex, array1d< globalIndex > > m_unmappedGlobalIndicesInNodelist;
+
+  /// Map of unmapped global indices in the element-to-face map
+  map< localIndex, array1d< globalIndex > > m_unmappedGlobalIndicesInEdgelist;
+
+  /// Map of unmapped global indices in the element-to-face map
+  map< localIndex, array1d< globalIndex > > m_unmappedGlobalIndicesInFacelist;
+
+  /// List of the fractured elements on this rank
+  SortedArray< localIndex > m_fracturedCells;
+
+  /// Map from local Cell Elements to Embedded Surfaces
+  EmbSurfMapType m_toEmbeddedSurfaces;
```

Comment:
Do you really need these for the `ParticleSubRegion`? Are there going to be fractured particles? I am not sure I can picture this.  `m_toEmbeddedSurfaces` is a list of the embeddedSurface elements inside each cellblock. I am not sure it has a meaning for particles.